pkg/server/plugin/upstreamauthority/spire/spire.go

```diff
@@ -100,11 +98,29 @@ func (m *spirePlugin) MintX509CA(ctx context.Context, request *upstreamauthority
 }
 
 func (m *spirePlugin) PublishJWTKey(ctx context.Context, req *upstreamauthority.PublishJWTKeyRequest) (*upstreamauthority.PublishJWTKeyResponse, error) {
-	return nil, makeError(codes.Unimplemented, "publishing upstream is unsupported")
-}
+	m.mtx.RLock()
+	defer m.mtx.RUnlock()
+
+	wCert, wKey, wBundle, err := m.getWorkloadSVID(ctx, m.config)
+	if err != nil {
+		return nil, err
+	}
+
+	conn, err := m.newNodeClientConn(ctx, wCert, wKey, wBundle)
+	if err != nil {
+		return nil, err
+	}
+	nodeClient := node.NewNodeClient(conn)
+	defer conn.Close()
 
-func makeError(code codes.Code, format string, args ...interface{}) error {
-	return status.Errorf(code, "upstreamauthority-spire: "+format, args...)
+	resp, err := nodeClient.PushJWTKeyUpstream(ctx, &node.PushJWTKeyUpstreamRequest{JwtKey: req.JwtKey})
```

Comment:
what do you think about all our errors become grpc status errors?

Reply:
This error will be a gRPC status error since that is what is returned from a gRPC client RPC call.

Reply:
> what do you think about all our errors become grpc status errors?

Sounds good

> This error will be a gRPC status error since that is what is returned from a gRPC client RPC call.

Thanks, will change the other ones